Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/questions-programming/questions-programming.html

```diff
@@ -4,58 +4,71 @@
-            <option>Option 1</option>
-            <option>Option 2</option>
-            <option>Option 3</option>
+          <select class="form-control half-width" id="categorySelect" (change)="selectCategory($event.target.value)" ngModel required name="categorySelect" #categorySelect="ngModel">
+            <option [disabled]="categorySelect.dirty" selected value="">Please select a category</option>
+            <option *ngFor="let category of categoryList">{{category.categoryName}}</option>
           </select>
-          <div class="errors-container"></div>
+          <div class="errors-container">
+            <div *ngIf="categorySelect.errors">
+              <div *ngIf="(categorySelect.dirty || categorySelect.touched)">
+                <p class="text-danger">Please select a category</p>
+              </div>
+            </div>
+          </div>
         </div>
         <div class="form-group">
           <label class="h4">Difficulty Level</label>
-          <select class="form-control small-width">
-            <option>Easy</option>
+          <select class="form-control small-width" name="difficulty" (change)="selectDifficulty($event.target.value)">
+            <option selected>Easy</option>
             <option>Medium</option>
             <option>Hard</option>
           </select>
-          <div class="errors-container"></div>
+        </div>
+        <div class="errors-container">
         </div>
         <div class="form-group">
           <label class="h4">Question Details</label>
-          <textarea class="form-control half-width" placeholder="Question details here" rows="3"></textarea>
-          <div class="errors-container"></div>
+          <ckeditor placeholder="Question details here" rows="3" [(ngModel)]="questionModel.question.questionDetail" id="detail" name="detail" #detail="ngModel" required></ckeditor>
+          <div class="errors-container">
+            <div *ngIf="detail.errors && (detail.dirty || detail.touched)">
+              <div [hidden]="!detail.errors.required">
+                <p class="text-danger">Question detail is required</p>
+              </div>
+            </div>
+          </div>
         </div>
         <div class="form-group">
           <label class="h4">Code Languages</label>
           <md-chip-list class="form-control half-width" (click)="autocomplete.focus()">
-            <md-chip>Objective C <em class="material-icons">close</em></md-chip>
-            <md-chip>C++ <em class="material-icons">close</em></md-chip>
+            <md-chip *ngFor="let selectedLanguage of selectedLanguageList">{{selectedLanguage}}<i class="material-icons" (click)="removeLanguage(selectedLanguage)">close</i></md-chip>
             <md-input-container>
-              <input mdInput placeholder="Code Languages" [mdAutocomplete]="code" #autocomplete>
+              <input mdInput placeholder="Code Languages" [mdAutocomplete]="code" #autocomplete ngModel name="languageInput" #languageInput="ngModel">
```

Comment:
Cannot remove it. 
`form.element.dirty` syntax is not working. So decided to stay with it.